test/functional/p2p_tx_download.py

```diff
@@ -204,17 +201,45 @@ def test_notfound_fallback(self):
             assert_equal(peer_fallback.tx_getdata_count, 0)
         peer_notfound.send_and_ping(msg_notfound(vec=[CInv(MSG_WTX, WTXID)]))  # Send notfound, so that fallback peer is selected
         peer_fallback.wait_until(lambda: peer_fallback.tx_getdata_count >= 1, timeout=1)
-        with p2p_lock:
-            assert_equal(peer_fallback.tx_getdata_count, 1)
 
-    def test_preferred_inv(self):
-        self.log.info('Check that invs from preferred peers are downloaded immediately')
-        self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+    def test_preferred_inv(self, preferred=False):
+        if preferred:
+            self.log.info('Check invs from preferred peers are downloaded immediately')
+            self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+        else:
+            self.log.info('Check invs from non-preferred peers are downloaded after {} s'.format(NONPREF_PEER_TX_DELAY))
+        mock_time = int(time.time() + 1)
+        self.nodes[0].setmocktime(mock_time)
         peer = self.nodes[0].add_p2p_connection(TestP2PConn())
         peer.send_message(msg_inv([CInv(t=MSG_WTX, h=0xff00ff00)]))
-        peer.wait_until(lambda: peer.tx_getdata_count >= 1, timeout=1)
+        peer.sync_with_ping()
+        if preferred:
+            peer.wait_until(lambda: peer.tx_getdata_count >= 1, timeout=1)
+        else:
+            with p2p_lock:
+                assert_equal(peer.tx_getdata_count, 0)
+            self.nodes[0].setmocktime(mock_time + NONPREF_PEER_TX_DELAY)
+            peer.wait_until(lambda: peer.tx_getdata_count >= 1, timeout=1)
+
+    def test_txid_inv_delay(self, glob_wtxid=False):
+        self.log.info('Check that inv from a txid-relay peers are delayed by {} s, with a wtxid peer {}'.format(TXID_RELAY_DELAY, glob_wtxid))
+        self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+        mock_time = int(time.time() + 1)
+        self.nodes[0].setmocktime(mock_time)
+        peer = self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=False))
+        if glob_wtxid:
+            # Add a second wtxid-relay connection otherwise TXID_RELAY_DELAY is waived in
+            # lack of wtxid-relay peers
+            self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=True))
+        peer.send_message(msg_inv([CInv(t=MSG_TX, h=0xff11ff11)]))
+        peer.sync_with_ping()
         with p2p_lock:
-            assert_equal(peer.tx_getdata_count, 1)
+            if glob_wtxid:
+                    assert_equal(peer.tx_getdata_count, 0)
+            else:
+                    assert_equal(peer.tx_getdata_count, 1)
```

Comment:
Tabs 4 spaces instead of 8, but you could also simplify here with
```diff
-            if glob_wtxid:
-                    assert_equal(peer.tx_getdata_count, 0)
-            else:
-                    assert_equal(peer.tx_getdata_count, 1)
+            assert_equal(peer.tx_getdata_count, 0 if glob_wtxid else 1)
```

Reply:
Less is better.